store/src/sqlite/model.rs

```diff
@@ -628,10 +631,13 @@ where
             .bind(right_fencepost.as_bytes())
             .fetch_all(self.pool.reader())
             .await?;
-        Ok(rows.first().map(|row| {
-            let bytes: Vec<u8> = row.get(0);
-            EventId::from(bytes)
-        }))
+        Ok(rows
+            .first()
+            .map(|row| {
+                let bytes: Vec<u8> = row.get(0);
+                EventId::try_from(bytes)
```

Comment:
Why not `EventId::try_from(bytes)?` instead of  `.transpose()?` after it is wrapped by the map?

Reply:
Because we are inside a closure of the Option.map call. So `?` does not early return from the outer function but only the closure, which means the closure needs to return Result<EventId> either way which gives us Option<Result<EventId>> and we need to transpose.